### PR TITLE
Adds more blueprint functionality

### PR DESCRIPTION
**Pull request by clusterfack**
Addresses #2914, adds renaming for several more machinery types that take their name from the area. 

Unfortunately does not work properly as it should be readding the vents/scrubbers to the new area as well as renaming all the machinery properly from space. If someone could let me know how I fucked up the procs I added to create area that they would not readd vents/scrubbers or rename that would be great because I've been banging my head against it for a few hours.

### CODE REVIEW

File: code/game/machinery/camera/camera.dm [hunk 1]
```diff
@@ -46,20 +46,23 @@ var/list/camera_names=list()
...
-			suffix++
+		name_camera()
 	..()
 
+/obj/machinery/camera/proc/name_camera()
+	var/area/A=get_area(src)
+	var/basename=A.name
+	var/nethash=english_list(network)
+	var/suffix = 0
+	while(1)
```
Comment (ComicIronic) [inline code/game/machinery/camera/camera.dm]: This could just be while(nethand+c_tag in camera_names). It has the same effect as the break, but saner.

**Comment by Probe1** [comment]
figure out how to blueprint the escape shuttle area without interfering with it arriving/leaving.

**Comment by clusterfack** [comment]
That would require magicks beyond your and my comprehension.

**Comment by ComicIronic** [comment]
I have a solution: DON'T BUILD OUT OF THE ESCAPE AIRLOCKS.

File: code/game/objects/items/blueprints.dm [hunk 2]
```diff
@@ -186,7 +189,26 @@ move an amendment</a> to the drawing.</p>
 			M.name = replacetext(M.name,oldtitle,title)
 		for(var/obj/machinery/door/M in RA)
 			M.name = replacetext(M.name,oldtitle,title)
-	//TODO: much much more. Unnamed airlocks, cameras, etc.
+		for(var/obj/machinery/light_switch/M in RA)
```
Comment (wild-billy) [inline code/game/objects/items/blueprints.dm]: yikes, this entire proc:
```dm
/obj/item/blueprints/proc/set_area_machinery_title(var/area/A,var/title,var/oldtitle)
 	if (!oldtitle) // or replacetext goes to infinite loop
 		return
 	for(var/area/RA in A.related)
 		for(var/obj/machinery/alarm/M in RA)
 			M.name = replacetext(M.name,oldtitle,title)
 		for(var/obj/machinery/power/apc/M in RA)
 			M.name = replacetext(M.name,oldtitle,title)
 		for(var/obj/machinery/atmospherics/unary/vent_scrubber/M in RA)
 			M.name = replacetext(M.name,oldtitle,title)
 		for(var/obj/machinery/atmospherics/unary/vent_pump/M in RA)
 			M.name = replacetext(M.name,oldtitle,title)
 		for(var/obj/machinery/door/M in RA)
 			M.name = replacetext(M.name,oldtitle,title)
	//TODO: much much more. Unnamed airlocks, cameras, etc.
		for(var/obj/machinery/light_switch/M in RA)
			M.name = replacetext(M.name,oldtitle,title)
		for(var/obj/machinery/firealarm/M in RA)
			M.name = replacetext(M.name,oldtitle,title)
		for(var/obj/machinery/camera/M in RA)
			var/oldname = M.name
			M.name = replacetext(M.name,oldtitle,title)
			if(M.name != oldname)
				M.name_camera()
```
Should be something like:
```dm
//TODO: Refactor this with a var and helper proc on machines
/var/global/list/areaRenamableMachines = list(
    /obj/machinery/alarm,
    /obj/machinery/power/apc,
    /obj/machinery/atmospherics/unary/vent_scrubber,
    /obj/machinery/atmospherics/unary/vent_pump,
    /obj/machinery/door,
    /obj/machinery/light_switch,
    /obj/machinery/firealarm)

/obj/item/blueprints/proc/set_area_machinery_title(var/area/A,var/title,var/oldtitle)
    if (!oldtitle) CRASH("set_area_machinery_title() fed null value for \"oldtitle\"")
    for(var/area/RA in A.related)
        for(var/obj/machinery/M in RA)
            if(M.type in renamableTypes)
                M.name = replacetext(M.name,oldtitle,title)
            else if(istype(M,/obj/machinery/camera) //TODO: align cameras to the rest of the machines
                var/oldname = M.name
                M.name = replacetext(M.name,oldtitle,title)
                if(M.name != oldname) M.name_camera()
```

Comment (clusterfack) [inline code/game/objects/items/blueprints.dm]: Just like of those things that just kept getting bigger I suppose

**Comment by clusterfack** [comment]
I had really ought to finish this

**Comment by Probe1** [comment]
lemme blueprint the shuttle.  comeon.  you know you can make it work.  you're the best.  you just gotta prove it.. with this!  yes, that's it.  i'm certain. ♫to be the very best like no one ever was♫

**Comment by SkowronX** [comment]
update when

**Comment by clusterfack** [comment]
Pipe code is hard as fuck to deal with in this specific case, and I really want for vents/scrubbers to become part of the new room with this PR.

**Comment by ComicIronic** [comment]
If I were you, I'd add a machine flag for area name dependancy and move area naming to a generic proc that machines could overwrite specifically, and then use that proc everywhere instead of copypasted code.

**Comment by clusterfack** [comment]
That was my plan when I saw it bloat before, work upon work to finish this.

**Comment by clusterfack** [comment]
Anyway I fixed this, it's done. I was hoping to add a capability to merge/separate rooms but nah fuck that.

**Comment by Probe1** [comment]
I hope one day because that would be a huge QoL improvement.

**Comment by clusterfack** [comment]
To do that I need some really specific circumstances to allow/disallow it, then it's just a matter of adopting the code that already exists for room detection.

**Pull request merged**